draft-marchan-kdl2.md

```diff
@@ -1013,12 +1102,27 @@ multi-line-raw-string-body :=
 // Numbers
 number := keyword-number | hex | octal | binary | decimal
 
-decimal := sign? integer ('.' integer)? exponent?
+decimal := significand exponent?
+suffixed-decimal := significand (
+    bare-type-suffix
+    | (exponent? explicit-type-suffix)
+)
+significand := sign? significand-initial integer? ('.' integer)?
 exponent := ('e' | 'E') sign? integer
 integer := digit (digit | '_')*
+significand-initial = digit
+    - '0b'
+    - '0o'
+    - '0x'
 digit := [0-9]
 sign := '+' | '-'
 
+bare-type-suffix := bare-type-suffix-initial identifier-char*
+bare-type-suffix-initial := identifier-char
+    - '.' - ','
+    - (('e' | 'E') (sign | digit))
+explicit-type-suffix := '#' identifier-string
```

Comment:
Similarly, this isn't a valid use of `-`, since it's not subtracting from `identifier-char`. Easier to fix, tho:

```suggestion
bare-type-suffix := (bare-type-suffix-initial identifier-char*) | 'e' | 'E'
bare-type-suffix-initial := identifier-char - ('.' | ',' | 'e' | 'E') | identifier-char{2} - (('e' | 'E') (sign | digit))
    - '.' - ','
    - (('e' | 'E') (sign | digit))
explicit-type-suffix := '#' identifier-string
```

This allows `1e` and `1egg`, but disallows `1e+`, `1e1`, and `1e+gg`.

Reply:
Dang it, this doens't work, the second clause in `bare-type-suffix-initial` lets `.f` smuggle in as valid. I'll need to tweak further.

Reply:
Okay, fixed, using the same techniques we use in idents to avoid the ambiguous numeric patterns.